x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MachineLearning.java

```diff
@@ -889,7 +889,7 @@ private static void reportClashingNodeAttribute(String attrName) {
 
     @Override
     public List<RescorerSpec<?>> getRescorers() {
-        if (enabled && machineLearningExtension.get().isLearningToRankEnabled()) {
+        if (enabled && machineLearningExtension.get() != null && machineLearningExtension.get().isLearningToRankEnabled()) {
```

Comment:
This fix for tests is not ideal. If production was ever changed so that rescorers were requested before the extension was set up then this test would just silently ignore the intention of the extension in production. Obviously the tests on the early commits on this PR prove this doesn't happen today, but we'll be vulnerable to somebody making a mistake in a future refactoring.

Which test needs this fix? Maybe there's a way to set up a dummy extension that makes that test happy.

Reply:
Hey @droberts195 I did revert this patch so we will be able to create a more targeted one.

Reply:
Most of the failure where caused by tests creating a search module using the module without calling loadExtensions. 

I did some changes in the tests.